MiniApp/Classes/core/RealMiniApp.swift

```diff
@@ -128,18 +130,31 @@ internal class RealMiniApp {
         }
     }
 
-    func getMiniAppView(appInfo: MiniAppInfo, queryParams: String? = nil, completionHandler: @escaping (Result<MiniAppDisplayDelegate, Error>) -> Void, messageInterface: MiniAppMessageDelegate? = nil, adsDisplayer: MiniAppAdDisplayer? = nil) {
-        DispatchQueue.main.async {
-            let miniAppDisplayProtocol = self.displayer.getMiniAppView(miniAppId: appInfo.id,
-                                                                       versionId: appInfo.version.versionId,
-                                                                       projectId: self.miniAppClient.environment.projectId,
-                                                                       miniAppTitle: appInfo.displayName ?? "Mini app",
-                                                                       queryParams: queryParams,
-                                                                       hostAppMessageDelegate: messageInterface ?? self,
-                                                                       adsDisplayer: adsDisplayer)
-            self.miniAppStatus.setDownloadStatus(true, appId: appInfo.id, versionId: appInfo.version.versionId)
-            self.miniAppStatus.setCachedVersion(appInfo.version.versionId, for: appInfo.id)
-            completionHandler(.success(miniAppDisplayProtocol))
+    func getMiniAppView(appInfo: MiniAppInfo,
+                        queryParams: String? = nil,
+                        completionHandler: @escaping (Result<MiniAppDisplayDelegate, Error>) -> Void,
+                        messageInterface: MiniAppMessageDelegate? = nil,
+                        adsDisplayer: MiniAppAdDisplayer? = nil) {
+        isRequiredPermissionsAllowed(
```

Comment:
This call to `isRequiredPermissionsAllowed` is buried inside several other function calls, so it makes it more difficult to understand the behavior when creating a Mini App. It seems like the code would be easier to read if this call was moved higher, i.e. into `createMiniApp`. But I'm not sure if you can keep the exact same behavior if you move it somewhere else?

Reply:
But I would suggest that we call this `isRequiredPermissionsAllowed` method right even before we download.

Reply:
I think that `isRequiredPermissionsAllowed` is unrelated to downloading, so it doesn't make sense to call it before download. For example, even if the mini app is already downloaded, we still need to check `isRequiredPermissionsAllowed` because the user could have revoked permissions in the permission settings screen.

Anyways, I think it's okay to leave this here - it might need a lot of refactoring to move it somewhere else.